Add some comparison with the 'naive approach' (#24)
LukeMathWalker/cargo-chef: README.md

We are using four stages: the first computes the recipe file, the second caches our dependencies, the third builds the binary and the fourth is our runtime environment.  
As long as your dependencies do not change the `recipe.json` file will stay the same, therefore the outcome of `cargo cargo chef cook --release --recipe-path recipe.json` will be cached, massively speeding up your builds (up to 5x measured on some commercial projects).

## Benefits vs Limitations

`cargo-chef` has been tested on a few OpenSource projects and some of commercial projects, but our testing has definitely not exhausted the range of possibilities when it comes to `cargo build` customisations and we are sure that there are a few rough edges that will have to be smoothed out - please file issues on [GitHub](https://github.com/LukeMathWalker/cargo-chef).

> #### Benefits of `cargo-chef`:

Reviewer: ```suggestion
### Benefits of `cargo-chef`:
```

> A common alternative is to load a minimal `main.rs` into a container with the cargo toml and lockfile to build a docker layer that consists of only your dependencies ([more info here](https://www.lpalmieri.com/posts/fast-rust-docker-builds/#caching-rust-builds)). This means changes to your actual code will only rebuild the code itself. This is fragile compared to `cargo-chef` which is more set-and-forget:

Reviewer: ```suggestion
A common alternative is to load a minimal `main.rs` into a container with `Cargo.toml` and `Cargo.lock` to build a Docker layer that consists of only your dependencies ([more info here](https://www.lpalmieri.com/posts/fast-rust-docker-builds/#caching-rust-builds)). This is fragile compared to `cargo-chef` which will instead:
```

> - automatically picks up multiple crates

Reviewer: ```suggestion
- automatically pick up all crates in a workspace (and new ones as they are added)
```

> - moving files or crates around is handled automatically

Reviewer: ```suggestion
- keep working when files or crates are moved around, which would instead require manual edits to the `Dockerfile` using the "manual" approach
```

> - for multiple workspaces fewer intermediary layers are generated

Reviewer: What do you mean here?

Author: Essentially for each crate in the workspace you will need some variation of

```Dockerfile
WORKDIR $(root)/path/to/crate
COPY ./path/to/crate/Cargo.toml ./path/to/crate/Cargo.lock ./
RUN mkdir -p src/ && echo "fn main() { }" > src/main.rs
RUN cargo build --release
```

to capture all the dependencies. This will cause your 'build' container to have lots and lots of layers any one of which can be invalidated individually (as well as worsening your signal-to-noise ratio). 

I could elaborate or omit it :)

Author: We can't help with cache invalidation because if your cargo file changes, it changes but we can get rid of the noise.

> #### Limitations and caveats:

Reviewer: ```suggestion
### Limitations and caveats:
```

> - for multiple workspaces fewer intermediary layers are generated

Reviewer: ```suggestion
- generate fewer intermediate Docker layers (for workspaces)
```
